src/Serializer/Serializer.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Answear\GlsBundle\Serializer;
+
+use Answear\GlsBundle\Request\RequestInterface;
+use Answear\GlsBundle\Serializer\Normalizer\EnumNormalizer;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Normalizer;
+use Symfony\Component\Serializer\Serializer as SymfonySerializer;
+
+class Serializer
+{
+    private const FORMAT = 'json';
+
+    private SymfonySerializer $serializer;
+
+    public function serialize(RequestInterface $request): string
+    {
+        return $this->getSerializer()->serialize(
+            $request,
+            self::FORMAT,
+            [Normalizer\AbstractObjectNormalizer::SKIP_NULL_VALUES => true]
+        );
+    }
+
+    public function decodeResponse(string $class, ResponseInterface $response)
+    {
+        return $this->getSerializer()->deserialize(
+            $response->getBody()->getContents(),
+            $class,
+            self::FORMAT
+        );
+    }
+
+    private function getSerializer(): SymfonySerializer
```

Comment:
Why a private method instead of a constructor?

Reply:
creating a Symfony Serializer on the constructor creates it on every request - unnecessary and redundant